LethalMon/Behaviours/MaskedTamedBehaviour.cs

```diff
@@ -0,0 +1,1125 @@
+        List<Material> originalMaskMaterials = new();
+        Transform? originalMaskParent = null;
+        Vector3 originalMaskLocalPosition = Vector3.zero;
+
+        float timeWearingMask = 0f;
+
+        Coroutine? maskTransferCoroutine = null;
+
+        GameObject? Mask => Masked?.maskTypes[Masked.maskTypeIndex];
+        Animator? _maskAnimator = null;
+        Animator? MaskAnimator
+        {
+            get
+            {
+                if (_maskAnimator == null)
+                    _maskAnimator = Mask?.GetComponent<Animator>();
+
+                return _maskAnimator;
+            }
+        }
+
+        Color? originalNightVisionColor = null;
+        float originalNightVisionIntensity = 366f;
+
+        public bool escapeFromBallEventRunning = false;
+
+        // Ghosts
+        internal bool isGhostified = false;
+        internal float ghostLifetime = 0f;
+        internal bool IsGhostAboutToDie => (MaximumGhostLifeTime - ghostLifetime) < 2f;
+        internal float GhostSpawnTime => 3.5f + 1f - (GhostChaseSpeed / 3f);
+        List<MaskedPlayerEnemy> spawnedGhostMimics = new List<MaskedPlayerEnemy>();
+        MaskedTamedBehaviour? parentMimic = null;
+        float glitchCooldown = 0f;
+
+        private Material? _ghostMaterial = null;
+        internal Material GhostMaterial
+        {
+            get
+            {
+                if (_ghostMaterial == null)
+                {
+                    _ghostMaterial = new Material(Utils.WireframeMaterial);
+                    _ghostMaterial.SetColor("_EdgeColor", new Color(0.8f, 0.9f, 1f, 0.15f));
+                    _ghostMaterial.SetFloat("_WireframeVal", 1f);
+                    _ghostMaterial.SetFloat("_MaxVisibilityDistance", 15f);
+                }
+
+                return _ghostMaterial;
+            }
+        }
+
+        private Material? _ghostEyesMaterial = null;
+        internal Material GhostEyesMaterial
+        {
+            get
+            {
+                if (_ghostEyesMaterial == null)
+                {
+                    _ghostEyesMaterial = new Material(Shader.Find("HDRP/Unlit"));
+                    _ghostEyesMaterial.color = new Color(0.8f, 0.9f, 1f);
+                }
+
+                return _ghostEyesMaterial;
+            }
+        }
+
+        // Audio
+        internal static AudioClip? ghostAmbientSFX = null, ghostHissSFX = null, ghostHissFastSFX = null, ghostPoofSFX = null;
+
+        // Audio (Far)
+        internal static AudioClip? ghostAmbientFarSFX = null;
+
+        internal static List<Tuple<AudioClip, AudioClip>> GhostVoices = new List<Tuple<AudioClip, AudioClip>>();
+        internal float timeTillNextGhostVoice = 0f;
+
+        internal AudioSource? farAudio = null;
+        #endregion
+
+        #region Cooldowns
+        private static readonly string CooldownId = "masked_lendmask";
+
+        internal override Cooldown[] Cooldowns => new[] { new Cooldown(CooldownId, "Lending mask", 2f) };
```

Comment:
Can you add a config field for the cooldown time and also for the view distance?

Reply:
Done